ckanext/datagovuk/tests/test_s3_resources_plugin.py

```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/env python3
+import cgi
+from StringIO import StringIO
+
+import botocore.session
+import pylons.config as config
+from botocore.stub import ANY, Stubber
+from ckanext.datagovuk.plugin import DatagovukPlugin
+from mock import patch
+from nose.tools import assert_raises, assert_raises_regexp
+
+
+class MockAction:
+    pkg = None
+    rsc = None
+
+    def __init__(self, _type):
+        self._type = _type
+
+    def __call__(self, *arg):
+        if self._type == "package_show":
+            return self.pkg or {"name": "test"}
+        elif self._type == "resource_show":
```

Comment:
I don't think we need a response for `resource_show` as its not called in `upload.py`